File: OnlineRepairBooking/backend/internal/routes/worker.go
package routes

import (
	"online-repair-booking/internal/handlers"
	"online-repair-booking/pkg/database"

	"github.com/labstack/echo/v4"
)

func RegisterWorkerRoutes(g *echo.Group) {
	db := database.MySQL
	workerHandler := handlers.NewWorkerHandler(db)

	workerGroup := g.Group("/workers")
	{
		workerGroup.GET("", workerHandler.GetWorkerList)
		workerGroup.GET("/:id", workerHandler.GetWorkerDetail)
		workerGroup.GET("/:id/reviews", workerHandler.GetWorkerReviews)
		workerGroup.POST("/register", workerHandler.WorkerRegister)
	}
}

File: OnlineRepairBooking/frontend/src/api/worker.js
import request from '@/utils/request'

export const getWorkerList = (params) => {
  return request({
    url: '/workers',
    method: 'get',
    params
  })
}

export const getWorkerDetail = (id) => {
  return request({
    url: `/workers/${id}`,
    method: 'get'
  })
}

export const getWorkerReviews = (workerId, params) => {
  return request({
    url: `/workers/${workerId}/reviews`,
    method: 'get',
    params
  })
}

export const workerRegister = (data) => {
  return request({
    url: '/workers/register',
    method: 'post',
    data
  })
}

export const getWorkerStats = () => {
  return request({
    url: '/worker/stats',
    method: 'get'
  })
}
